### PR TITLE
zlib: update to 1.2.13

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: subprojects/packagefiles/zlib/meson.build [hunk 1]
```diff
@@ -16,9 +16,9 @@ if cc.get_argument_syntax() == 'msvc'
 else
   # Don't spam consumers of this wrap with these warnings
   compile_args += cc.get_supported_arguments('-Wno-implicit-fallthrough', '-Wno-implicit-function-declaration')
-  if cc.get_id() == 'gcc' and host_machine.system() != 'windows'
-    vflag = '-Wl,--version-script,@0@/zlib.map'.format(meson.current_source_dir())
-    link_args += [vflag]
+  gnu_sym_ldflag = '-Wl,--version-script,@0@/zlib.map'.format(meson.current_source_dir())
+  if get_option('buildtype') != 'static' and cc.links('', name: '-Wl,--version-script', args: ['-shared', gnu_sym_ldflag])
```
Comment (eli-schwartz) [inline subprojects/packagefiles/zlib/meson.build:20]: You don't need to check the buildtype, link_args will simply do nothing whatsoever if the linker is never getting run?

Comment (eli-schwartz) [reply subprojects/packagefiles/zlib/meson.build:20]: Also what buildtype is this supposed to be, presumably default_library?

Comment (neheb) [reply subprojects/packagefiles/zlib/meson.build:20]: copy/paste from libsndfile. looks like that's wrong too...

**Comment by neheb** [comment]
@jpakkane this is ready.

**Comment by neheb** [comment]
wonder if it's better to use https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerget_supported_link_arguments actually...

**Pull request merged**